src/python/gem5/components/boards/abstract_board.py

```diff
@@ -256,6 +258,9 @@ def set_workload(self, workload: WorkloadResource) -> None:
 
         func(**workload.get_parameters())
 
+    def get_workload(self) -> WorkloadResource:
```

Comment:
```suggestion
    def get_workload(self) -> Optional[WorkloadResource]:
```

This could be `None` in the case a workload is not set.